SoundCrate/scfront
# [Jem] Adding UX/UI Options to Lists and the Profile Page

## Context
- Users can create/delete song lists and add/remove songs to those lists. 

## Changes
- Created more API routes regarding editing of lists and contents
- Updated modals to disappear after use and added modals for better UX/UI functionality of lists
- Ensured that any updates made reflect on pages without refreshing.
- Added messages for missing or empty content

## Testing For UX/UI Updates
1. Create a new account.
2. Ensure that the no reviews, no albums, and no lists messages appear.
3. Navigate to the different sub-pages within a profile page (reviews, lists, albums). Ensure that the no reviews, no albums, and no lists messages appear.

## Testing For Lists
1. Create a new account.
5. Throughout all steps below: Check MongoDB to confirm that lists are being updated accordingly. 
6. Create lists from both the profile and lists page (with and without descriptions). Make sure that the modal disappears automatically and lists appear without refreshing.
7. Delete lists and lists page. Make sure that the modal disappears automatically and lists delete without refreshing.
8. Navigate to the contents of a single list. Edit List Details by changing the title and description.
9. Navigate to the contents of a single list. Delete songs and ensure that the list updates without refreshing.
10. Ensure that there is no access to edit, update, or delete the contents of another user's lists.

## ToDo - IMPORTANT FOR EVERYONE
- Add protection against editing lists to have no title.
- Button and overall design of components across the profile page require refactoring (Following vs Followers vs Reviews vs Lists) so they match
- No Albums on the Albums page need to show "No Albums" instead of "Loading"
- Modals for other components (following, albums, reviews) need to disappear automatically and the page must reflect the changes without refresh.
- Reviews should be able to be deleted (and possible editing if we have time)

<img width="789" alt="blank" src="https://github.com/SoundCrate/scfront/assets/93406637/859080bf-dbf7-41d8-b076-abba644202f1">
<img width="827" alt="lists page" src="https://github.com/SoundCrate/scfront/assets/93406637/938c9a78-ca0c-4f1d-a7a1-ca984d8121be">
<img width="793" alt="within a list" src="https://github.com/SoundCrate/scfront/assets/93406637/ef6aed73-f291-4d4c-b2ea-8493d2233746">
